tests/test_install_mender_sh.py

```diff
@@ -225,25 +225,114 @@ def test_configure(
         check_installed(generic_debian_container, "mender-connect", installed=False)
         check_installed(generic_debian_container, "mender-configure")
 
-    def test_upgrade_all_packages(
+    def test_upgrade_mender_v3_series(
         self, generic_debian_container,
     ):
-        # Install latest stable software
+        # Install latest stable software (client v3)
         generic_debian_container.run(
             f"curl http://{SCRIPT_SERVER_ADDR}:{SCRIPT_SERVER_PORT}/install-mender.sh | bash -s"
         )
 
-        # And now upgrade to the freshly built packages
+        # Now upgrade to the freshly built packages
         local_apt_repo_from_built_packages(generic_debian_container)
-        generic_debian_container.run("apt -y upgrade")
+        generic_debian_container.run("apt --assume-yes upgrade")
 
-        # All packages should be upgraded
+        # Only mender-client should be upgraded
         check_installed(generic_debian_container, "mender-client")
+        check_installed(generic_debian_container, "mender-client4", installed=False)
+        check_installed(generic_debian_container, "mender-update", installed=False)
+        check_installed(generic_debian_container, "mender-auth", installed=False)
+        check_installed(generic_debian_container, "mender-flash", installed=False)
+        check_installed(generic_debian_container, "mender-setup", installed=False)
+        check_installed(generic_debian_container, "mender-snapshot", installed=False)
+        # The addons should not be removed
+        check_installed(generic_debian_container, "mender-connect")
+        check_installed(generic_debian_container, "mender-configure")
+
+    def test_upgrade_mender_v4_series_meta_package_with_addons(
+        self, generic_debian_container,
+    ):
+        # Install latest stable software (client v3)
+        generic_debian_container.run(
+            f"curl http://{SCRIPT_SERVER_ADDR}:{SCRIPT_SERVER_PORT}/install-mender.sh | bash -s"
+        )
+
+        local_apt_repo_from_built_packages(generic_debian_container)
+
+        # MEN-7010
+        # Currently, production mender-configure still depends (only) on mender-client.
+        # We need first to update this one to the freshly built in order to make the rest of the
+        # test succeed. This part can be removed after MEN-7010 is finished.
+        generic_debian_container.run("apt install --assume-yes mender-configure")
+
+        # Now install freshly built mender-client4
+        generic_debian_container.run(
+            "DEBIAN_FRONTEND=noninteractive apt install --assume-yes mender-update mender-client4"
```

Comment:
So it works fully automatic! Awesome!

Reply:
Well, almost :sweat_smile: - See reply in comment below.